pkg/data/data.go

```diff
@@ -153,6 +259,64 @@ func (newJob *NewJob) Validate(update bool) map[string]string {
 	return errs
 }
 
+type NewRole struct {
+	Name     string `form:"name"`
+	Email    string `form:"email"`
+	Phone    string `form:"phone"`
+	Role     string `form:"role"`
+	Resume   string `form:"resume"`
+	Linkedin string `form:"linkedin"`
+	Website  string `form:"website"`
+	Github   string `form:"github"`
+	CompLow  string `form:"complow"`
+	CompHigh string `form:"comphigh"`
+}
+
+func (newJob *NewRole) Validate(update bool) map[string]string {
+	errs := make(map[string]string)
+
+	if newJob.Name == "" {
+		errs["name"] = ErrNoName
+	}
+
+	if !update {
+		if newJob.Email == "" {
+			errs["email"] = ErrNoEmail
+		} else if _, err := mail.ParseAddress(newJob.Email); err != nil {
+			// TODO: Maybe do more than just validate email format?
+			errs["email"] = ErrInvalidEmail
+		}
+	}
+
+	if newJob.Role == "" {
+		errs["role"] = ErrNoRole
+	}
+
+	if newJob.Resume == "" {
+		errs["resume"] = ErrNoResume
+	}
+
+	if newJob.Linkedin != "" {
+		if _, err := url.ParseRequestURI(newJob.Linkedin); err != nil {
+			errs["linkedin"] = ErrInvalidUrl
+		}
+	}
+
+	if newJob.Website != "" {
+		if _, err := url.ParseRequestURI(newJob.Website); err != nil {
+			errs["website"] = ErrInvalidUrl
+		}
+	}
+
+	if newJob.Github != "" {
+		if _, err := url.ParseRequestURI(newJob.Github); err != nil {
+			errs["github"] = ErrInvalidUrl
+		}
+	}
+
+	return errs
```

Comment:
Thoughts on adding a validation for both compensation fields to be present? Either both missing, or both provided.